pkg/metrics/reconciliation.go

```diff
@@ -0,0 +1,50 @@
+package metrics
+
+import (
+	"fmt"
+
+	"github.com/kyma-incubator/reconciler/pkg/cluster"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+)
+
+// ReconciliationWaitingCollector provides number of clusters waiting to be reconciled:
+// - reconciler_reconciliation_waiting_total - total number of clusters waiting to be reconciled
+type ReconciliationWaitingCollector struct {
+	inventory cluster.Inventory
+	logger    *zap.Logger
+
+	waitingClustersDesc *prometheus.Desc
+}
+
+func NewReconciliationWaitingCollector(inventory cluster.Inventory, logger *zap.Logger) *ReconciliationWaitingCollector {
+	return &ReconciliationWaitingCollector{
+		inventory: inventory,
+		logger:    logger,
+		waitingClustersDesc: prometheus.NewDesc(prometheus.BuildFQName("", prometheusSubsystem, "reconciliation_waiting_total"),
+			"Total number of clusters waiting to be reconciled",
+			[]string{},
+			nil),
+	}
+}
+
+func (c *ReconciliationWaitingCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.waitingClustersDesc
+}
+
+// Collect implements the prometheus.Collector interface.
+func (c *ReconciliationWaitingCollector) Collect(ch chan<- prometheus.Metric) {
+	clusters, err := c.inventory.ClustersToReconcile()
```

Comment:
Arg.. I think we have to add another method the inventory. This will return only the clusters which have to be scheduled for reconciliation.

What we need is a list of all clusters which are currently reconciling or in an error / failed state.

Let's quickly sync tomorrow, but I think only the method-name has to be adjusted in your code. The rest should still work as before.